### PR TITLE
Submitted the solutions of week01's problems.

### PR DESCRIPTION
**Pull request by saifuzzaman-saif**
Solved the following problems in this PR.

Two Sum
Best Time to Buy and Sell Stock
Move Zeroes
Majority Element
Merge Sorted Array
Single Number
Trapping Rain Water
First Missing Positive
Best Time to Buy and Sell Stock - 2
Three Sum
Four Sum
Next Permutation
Merge Intervals
Meeting Rooms 2
Kth Largest Element in an Array
Insert,Delete,GetRandom in O(1)
Product of Array Except Itself
Longest Consecutive Sequence

### CODE REVIEW

File: week01/BestTimeToBuyAndSellStock.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Solution {
+
+    // TC = O(n) : where n is the size of Prices array.
+    // MC = O(1)
+
+    int inf = 1_000_000_000;
```
Comment (Sadman007) [inline week01/BestTimeToBuyAndSellStock.java]: Bad variable naming.
Should be static and capital.

File: week01/InsertDeleteGetRandom.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
+class RandomizedSet {
+
+    // TC = O(1) in average for each function call.
+    // MC = O(n) ; where n is the number calls will be made.
+    
+    List<Integer> vals;
+    Map<Integer, Integer> pos;
```
Comment (Sadman007) [inline week01/InsertDeleteGetRandom.java:7]: Map does not support constant operation in Java. You should have used HashMap.

**Review by Sadman007 — APPROVED** [review]
Hi @saifuzzaman-saif 

Nice work. Feel free to merge this PR with master.

File: week01/BestTimeToBuyAndSellStock.java (hunk 1 above)
Comment (saifuzzaman-saif) [reply week01/BestTimeToBuyAndSellStock.java]: Fixed it.

**Comment by saifuzzaman-saif** [comment]
@Sadman007  vai thanks a lot and merged it after resolving the comment.

**Pull request merged**